WebApplication1/WebApplication1/Controllers/UsersController.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using WebApplication1.Data;
+using WebApplication1.DTOs;
+using WebApplication1.Exceptions;
+using WebApplication1.Middleware;
+using WebApplication1.Models;
+
+namespace WebApplication1.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UsersController : ControllerBase
+    {
+        private readonly DataContext _context;
+        private readonly IConfiguration _configuration;
+
+        public UsersController(DataContext context,IConfiguration configuration ) {
+            _context = context;
+            _configuration = configuration;
+    }
+
+        [HttpPost]
+        [Route("Register")]
+        public IActionResult Register(UserDTO userDTO)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
```

Comment:
create a validator which checks the model state, this way you avoid a lot of code duplication